internal/pkg/pkg.go

```diff
@@ -0,0 +1,91 @@
+// Copyright 2020 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Package pkg defines the concept of a kpt package.
+package pkg
+
+import (
+	"os"
+	"path/filepath"
+
+	"github.com/GoogleContainerTools/kpt/internal/pipeline"
+	"github.com/GoogleContainerTools/kpt/pkg/kptfile"
+)
+
+// pkg represents the static metadata for a kpt package.
+type pkg struct {
+	// Absolute unique OS-defined path to the package directory.
+	UniquePath string
+
+	// Relative slash-separated path to the package directory.
+	// This is not guaranteed to be unique (e.g. in presence of symlinks) and should only
+	// be used for display purposes.
+	DisplayPath string
+
+	// A package can contain zero or one Kptfile meta resource.
+	// A nil value represents an implicit package.
+	Kptfile *kptfile.KptFile
+
+	// A package can contain zero or one Pipeline meta resource.
+	Pipeline *pipeline.Pipeline
+}
+
+// New returns a pkg given an absolute or relative OS-defined path.
+// Use ReadKptfile or ReadPipeline on the return value to read meta resources from filesystem.
+func New(path string) (pkg, error) {
```

Comment:
We should be returning `*pkg` here or you are thinking something else here ?

Reply:
Done. Are you concerned about passing large values? Currently, this struct contains couple of string values and everything else is a pointer, so it's not a large struct. But it may expand in the future.